yarn-project/aztec.js/src/contract/contract_function_interaction.ts

```diff
@@ -103,13 +103,20 @@ export class ContractFunctionInteraction extends BaseContractInteraction {
     const txRequest = await this.create();
     const simulatedTx = await this.wallet.simulateTx(txRequest, true, options?.from, options?.skipTxValidation);
 
-    // As account entrypoints are private, for private functions we retrieve the return values from the first nested call
-    // since we're interested in the first set of values AFTER the account entrypoint
-    // For public functions we retrieve the first values directly from the public output.
-    const rawReturnValues =
-      this.functionDao.functionType == FunctionType.PRIVATE
```

Comment:
This assumed that we always simulate via an account contract and if we did not it would fail with undefined. Since I did exactly that in this PR I've stumbled upon this. Note that there is a [new issue](https://github.com/AztecProtocol/aztec-packages/issues/10631) that needs to be tackled.